src/printer.rs

```diff
@@ -107,17 +110,57 @@ impl<'a, T: Read> BinaryGuard<'a, T> {
     }
 }
 
+pub struct FormatOptions {
+    pub json_indent: usize,
+    pub json_record_separator: String
+}
+
+impl FormatOptions {
+    pub fn from_string(options: Option<String>) -> FormatOptions {
+        let mut format_options = FormatOptions::default();
+
+        // generate a map of the specified options
+        if let Some(options) = options {
+            let mut arguments = HashMap::new();
+            for argument in options.split(",") {
+                let argument_parts = argument.split(":").collect::<Vec<&str>>();
+                // no proper argument provided, thus skipping
+                if argument_parts.len() < 2 { continue; }
+                arguments.insert(argument_parts[0], argument_parts[1]);
+            }
+
+            // save the specified options to the `FormatOptions` struct
+            if let Some(json_indent) = arguments.get("json.indent") {
+                if let Ok(json_indent) = json_indent.parse::<usize>() {
+                    format_options.json_indent = json_indent;
+                }
+            }
+            if let Some(record_separator) = arguments.get("json.record.separator") {
+                format_options.json_record_separator = record_separator.to_string();
+            }
+        }
+        return format_options
+    }
+}
+
+impl Default for FormatOptions {
+    fn default() -> Self {
+        Self { json_indent: DEFAULT_JSON_INDENT, json_record_separator: String::from(DEFAULT_JSON_RECORD_SEPARATOR) }
```

Comment:
Since these values are not used anywhere, I would prefer to inline them within this function.